psa-crypto/src/operations/key_management.rs

```diff
@@ -307,3 +307,99 @@ fn complete_new_key_operation(
+/// #         permitted_algorithms: AsymmetricSignature::RsaPkcs1v15Sign {
+/// #             hash_alg: Hash::Sha256.into(),
+/// #         }.into(),
+/// #     },
+/// # };
+/// psa_crypto::init().unwrap();
+/// let my_key = key_management::generate(attributes, None).unwrap();
+/// let my_key_copy = key_management::copy_key(my_key, attributes, None);
+/// ```
+pub fn copy_key(key_id_to_copy: Id, attributes: Attributes, id: Option<u32>) -> Result<Id> {
+    initialized()?;
+    let key_handle_to_copy = key_id_to_copy.handle()?;
+
+    let mut key_attributes = psa_crypto_sys::psa_key_attributes_t::try_from(attributes)?;
+    let id = if let Some(id) = id {
+        unsafe { psa_crypto_sys::psa_set_key_id(&mut key_attributes, id) };
+        id
+    } else {
+        0
+    };
+    let mut handle_for_new_key = 0;
+
+    let copy_res = Status::from(unsafe {
+        psa_crypto_sys::psa_copy_key(key_handle_to_copy, &key_attributes, &mut handle_for_new_key)
+    })
+    .to_result();
+    Attributes::reset(&mut key_attributes);
+    copy_res?;
+    complete_new_key_operation(attributes.lifetime, id, handle_for_new_key)
+}
+
+/*  NOTE: This operation is not yet supported by Mbed Crypto. Once it is supported, uncomment and remove this notice
+/// Remove non-essential copies of key material from memory
+///
+/// An implementation is permitted to make additional copies of key material For keys that have been created with the `cache` policy enabled, an implementation is permitted to make additional copies of the key material that are not in storage and not for the purpose of ongoing operations.
+///
+/// This function will remove these extra copies of the key material from memory.
+///
+/// This function is not required to remove key material from memory in any of the following situations:
+///
+///     The key is currently in use in a cryptographic operation.
+///     The key is volatile.
+///
+/// # Example
+///
+/// ```
+/// # use psa_crypto::operations::key_management;
+/// # use psa_crypto::types::key::{Attributes, Type, Lifetime, Policy, UsageFlags};
+/// # use psa_crypto::types::algorithm::{AsymmetricSignature, Hash};
+/// # let mut attributes = Attributes {
+/// #     key_type: Type::RsaKeyPair,
+/// #     bits: 1024,
+/// #     lifetime: Lifetime::Volatile,
+/// #     policy: Policy {
+/// #         usage_flags: UsageFlags {
+/// #             cache: true,
+/// #             ..Default::default()
+/// #         },
+/// #         permitted_algorithms: AsymmetricSignature::RsaPkcs1v15Sign {
+/// #             hash_alg: Hash::Sha256.into(),
+/// #         }.into(),
+/// #     },
+/// # };
+/// psa_crypto::init().unwrap();
+/// let my_key = key_management::generate(attributes, None).unwrap();
+/// let size = key_management::purge_key(my_key).unwrap();
+/// ```*/
+/*
+pub fn purge_key(key_id: Id) -> Result<()> {
+    initialized()?;
+    let handle = key_id.handle()?;
+    Status::from(psa_crypto_sys::psa_purge_key(handle)).to_result()
+}*/
```

Comment:
Maybe it is better to still keep the function but have an empty implementation for now? Using a log to notice it, with the same comment that you have above.

Reply:
I've commented out the code and the example so now all it does is use the `error` macro. Is this what you meant?